core/core/src/observability/metrics.rs

```diff
@@ -0,0 +1,163 @@
+/// Public module so that the macro can access it, but should not be used outside of the parent module.
+pub mod __private {
+    use std::collections::HashMap;
+
+    use serde::Serialize;
+
+    use tracing_subscriber::fmt::MakeWriter;
+
+    #[derive(Debug, Serialize)]
+    #[serde(tag = "event_type")]
+    pub enum Event<'a> {
+        #[serde(rename = "SDKInit")]
+        SdkInit {
+            occurred_at: &'a str,
+            configuration_hash: Option<&'a str>,
+            data: SdkInitData<'a>
+        },
+        Metrics {
+            occurred_at: &'a str,
+            configuration_hash: Option<&'a str>,
+            data: MetricsData<'a>
+        },
+        // ProviderChange {
+        //     data: ()
+        // }
```

Comment:
this won't be needed, as we will not have failover directly in OneClient. Well it proves good idea.